### ISSUE TITLE
vagrant ssh -c no longer works properly on multi-word commands in vagrant 2.2.3

### ISSUE THREAD

**Issue by edmcman** [hashicorp/vagrant#10696]
### Vagrant version
2.2.3

### Host operating system
Ubuntu 14.04

### Guest operating system
Ubuntu 14.04

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|
  config.vm.box = 'sei'
  config.vm.provider :vsphere do |vsphere|
    vsphere.template_name = 'packer-templates/ubufuzz-cli-amd64-vmware-sas'
    vsphere.cpu_count = 8
    vsphere.memory_mb = 16000
  end

end
```

### Debug output
https://gist.github.com/edmcman/5db4fbcefbd5137bc3e45da6a0eb757e

### Expected behavior
Commands should execute inside the VM

### Actual behavior
The command was misparsed and part of the command was interpreted as a machine name

### Steps to reproduce
1. vagrant up
2. vagrant ssh -c "echo hello world"

### Notes
1. One word commands work okay:
```
fuzz@UbuFuzz64:~/vagrant/experiments/temp$ vagrant ssh -c "uptime"
 14:17:55 up 5 min,  1 user,  load average: 0.01, 0.18, 0.13
Connection to 172.16.41.134 closed.
```
2. The following works:
```
fuzz@UbuFuzz64:~/vagrant/experiments/temp$ vagrant ssh default -- "echo hello world"                     
hello world
```

**Comment by briancain** [comment]
@edmcman - Hmm..was step 2 the exact command you ran to get the error? It seems to work fine for me:

```
vagrant@vagrant:~/sshtest$ vagrant ssh -c "echo hello world"
hello world
Connection to 127.0.0.1 closed.
vagrant@vagrant:~/sshtest$ vagrant -v
Vagrant 2.2.3
```

It dose look like in the debug log that it's separating all of the command string into their own strings for some reason

```ruby
 INFO cli: CLI: [] "ssh" ["--command", "echo", "hello", "world", "default"]
```

Where as I don't see that when I test it :thinking: 

```ruby
 INFO cli: CLI: [] "ssh" ["--command", "echo hello world"]
```

I also don't have the "default" argument attached to my ssh command when running `vagrant ssh -c "echo hello world"`

**Comment by edmcman** [comment]
Here's what I get:

```
fuzz@UbuFuzz64:~/vagrant/experiments/temp$ vagrant ssh -c "echo hello world"
The machine with the name 'hello' was not found configured for
this Vagrant environment.
```

I recently upgraded from an ancient version of Vagrant where -c worked fine.

**Comment by chrisroberts** [comment]
@edmcman What shell are you using?

**Comment by edmcman** [comment]
@chrisroberts fuzz@UbuFuzz64:~$ bash --version
GNU bash, version 4.3.11(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

**Comment by edmcman** [comment]
I am doing a little bit of debugging.  I extracted the AppImage and ran with `strace -f`.  The first line from `strace` is normal:

`execve("./AppRun", ["./AppRun", "ssh", "-c", "echo hello world"], [/* 32 vars */]) = 0`

However, the next one is wrong:

`execve("/home/fuzz/vagrant/experiments/mayhem-rop/squashfs-root/usr/bin//vagrant", ["vagrant", "ssh", "-c", "echo", "hello", "world"], [/* 41 vars */]) = 0`

Looking at `AppRun`, it uses `/bin/sh`, which for me is `/bin/dash` version 0.5.7-4ubuntu1.

If I change the last line of `AppRun` from `exec ${EXEC} $@` to `exec ${EXEC} "$@"` parsing seems to be correct.

**Comment by ladar** [comment]
With the recent upload troubles, I decided to add testcases to my Robox check script. In the past the script was simple. It just verified that `vagrant up` and `vagrant ssh` worked. But with the recent `vagrant upload` troubles, I decided to add specific test cases, including test cases which tried to use `vagrant ssh` to send commands to a box, and check the result. When I wrote the test cases I ran into problems, the biggest being an issue I think is related to this bug. Specifically with the `generic/alpine35` box, it seemed to struggle with the syntax:

```
vagrant ssh --command 'echo ${SHELL}'
vagrant ssh --command "bash -c 'exit 0'"
```

It seemed that this exact command worked, sometimes, if I ran it manually from the command line, but almost always failed, or got hung, when it was run from inside my test script. It was strange, because only the Alpine 3.5 box seemed to consistently have issues with these commands. 

I had much better luck when I switched to the syntax:

```
vagrant ssh -- exit 0
vagrant ssh -- echo "\$SHELL" 
```

The Robox test script isn't part of the public repo, but I can provide more details, or create a sanitized version of it, if required.

The host in this case was a `CentOS 7.6.1810` robot... and I tested using the Virtualbox and libvirt plugins (my test robot doesn't have a VMWare plugin license).
